apps/chat/src/components/Header/Header.tsx

```diff
@@ -53,20 +65,62 @@ const Header = () => {
   );
 
   const handleToggleChatbar = useCallback(() => {
-    if (!showChatbar && isSmallScreen()) {
+    if (!showChatbar && isMediumScreen()) {
       dispatch(UIActions.setShowPromptbar(false));
+    }
+
+    if (!showChatbar && isSmallScreen()) {
       dispatch(UIActions.setIsProfileOpen(false));
     }
+
+    if (!showChatbar && !isMediumScreen()) {
+      if (!windowWidth) return;
+      const calculatedChatWidth = centralChatWidth({
+        oppositeSidebarWidth: promptbarWidth,
+        windowWidth,
+        currentSidebarWidth: chatbarWidth,
+      });
+
+      if (calculatedChatWidth < CENTRAL_CHAT_MIN_WIDTH) {
+        const newPromptbarWidth = getNewSidebarWidth({
+          windowWidth,
+          oppositeSidebarWidth: chatbarWidth,
+        });
+        dispatch(UIActions.setPromptbarWidth(newPromptbarWidth));
+      }
+    }
+
     dispatch(UIActions.setShowChatbar(!showChatbar));
-  }, [dispatch, showChatbar]);
+  }, [chatbarWidth, dispatch, promptbarWidth, showChatbar, windowWidth]);
 
   const handleTogglePromtbar = useCallback(() => {
-    if (!showPromptbar && isSmallScreen()) {
+    if (!showPromptbar && isMediumScreen()) {
       dispatch(UIActions.setShowChatbar(false));
+    }
+
+    if (!showPromptbar && isSmallScreen()) {
       dispatch(UIActions.setIsProfileOpen(false));
     }
+
+    if (!showPromptbar && !isMediumScreen()) {
```

Comment:
```suggestion
    if (!showPromptbar && !isMediumScreen() && showChatbar) {
```

Reply:
It is needed to reduce Chatbar width even if it is not visible. Othewise, it will be "jumping" if to open Chatbar and close-open Promtbar.